File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.LinkedList;
import java.util.Queue;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]); // 총 인원 n명
        int k = Integer.parseInt(input[1]); // 제거될 순서 k
        Queue<Integer> queue = new LinkedList<>();
        Queue<Integer> answer = new LinkedList<>();
        for (int i = 0; i < n; i++) queue.add(i + 1);
        int count = 0;
        while (answer.size() < n) {
            int temp = queue.poll();
            count++;
            if (count % k == 0) answer.add(temp);
            else queue.add(temp);
        }
        StringBuilder sb = new StringBuilder();
        sb.append("<");
        count = 0;
        while (count < n - 1) {
            sb.append(answer.poll() + ", ");
            count++;
        }
        sb.append(answer.poll() + ">");
        System.out.println(sb.toString());
    }
}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.kt
import java.io.BufferedReader

fun main() = with(BufferedReader(System.`in`.bufferedReader())) {
    val (n, k) = readLine().split(" ")

    val data = IntArray(n.toInt())

    var result = "<"
    var circle = 0
    var count = 0
    var pointer = 0
    while (count != n.toInt()) {
        if (data[pointer] == 0) {
            circle++
        }
        if (circle == k.toInt()) {
            data[pointer] = 1
            count++
            circle = 0
            result += "${pointer + 1}, "
        }
        if (pointer == n.toInt() - 1) {
            pointer = 0
        } else {
            pointer++
        }
    }
    println(result.substring(0, result.length - 2) + ">")
}